Packet++/header/SomeIpLayer.h

```diff
@@ -0,0 +1,361 @@
+	 * @param[in] methodID methodID to set
+	 */
+	void setMethodID(uint16_t methodID);
+
+	/**
+	 * Get the Length Field of the SOME/IP header
+	 * @return uint32_t The length field of the SOME/IP header
+	 */
+	uint32_t getLengthField() const;
+
+	/**
+	 * Get the requestID
+	 * @return uint32_t returned in host endian
+	 */
+	uint32_t getRequestID() const;
+
+	/**
+	 * Set the Request ID
+	 * @param[in] requestID requestID to set
+	 */
+	void setRequestID(uint32_t requestID);
+
+	/**
+	 * Get the sessionID
+	 * @return uint16_t returned in host endian
+	 */
+	uint16_t getSessionID() const;
+
+	/**
+	 * Set the Session ID
+	 * @param[in] sessionID sessionID to set
+	 */
+	void setSessionID(uint16_t sessionID);
+
+	/**
+	 * Get the clientID
+	 * @return uint16_t returned in host endian
+	 */
+	uint16_t getClientID() const;
+
+	/**
+	 * Set the Client ID
+	 * @param[in] clientID clientID to set
+	 */
+	void setClientID(uint16_t clientID);
+
+	/**
+	 * Get the protocolVersion
+	 * @return uint8_t
+	 */
+	uint8_t getProtocolVersion() const;
+
+	/**
+	 * Set the Protocol Version
+	 * @param[in] version version to set
+	 */
+	void setProtocolVersion(uint8_t version);
+
+	/**
+	 * Get the interfaceVersion
+	 * @return uint8_t
+	 */
+	uint8_t getInterfaceVersion() const;
+
+	/**
+	 * Set the Interface Version
+	 * @param[in] version version to set
+	 */
+	void setInterfaceVersion(uint8_t version);
+
+	/**
+	 * Get the message type
+	 * @return uint8_t
+	 */
+	uint8_t getMessageTypeAsInt() const;
+
+	/**
+	 * Get the message type
+	 * @return SomeIpLayer::MsgType
+	 */
+	SomeIpLayer::MsgType getMessageType() const;
+
+	/**
+	 * Set the Message Type
+	 * @param[in] type Type to set
+	 */
+	void setMessageType(MsgType type);
+
+	/**
+	 * Set the Message Type
+	 * @param[in] type Type to set
+	 */
+	void setMessageType(uint8_t type);
+
+	/**
+	 * Get the returnCode
+	 * @return uint8_t
+	 */
+	uint8_t getReturnCode() const;
+
+	/**
+	 * Set the returnCode
+	 * @param[in] returnCode ReturnCode to set
+	 */
+	void setReturnCode(uint8_t returnCode);
+
+	/**
+	 * Set the length field of the SOME/IP header
+	 * @param[in] payloadLength Length of the payload
+	 */
+	void setPayloadLength(uint32_t payloadLength);
+
+	/**
+	 * Get the Header Length With Payload, as is stated in the length field of the SOME/IP header.
+	 * @return uint32_t Length of the complete SOME/IP Pdu, including header and payload
+	 */
+	uint32_t getHeaderLengthWithPayload() const;
+
+	/**
+	 * @return A pointer for the layer payload, meaning the first byte after the header
+	 */
+	uint8_t *getPduPayload() const { return m_Data + sizeof(someiphdr);	}
+
+	/**
+	 * @return The size in bytes of the payload
+	 */
+	size_t getPduPayloadSize() const { return getHeaderLengthWithPayload() - sizeof(someiphdr); }
+
+	/**
+	 * Get the Length of the SOME/IP header inc payload
+	 * @return size_t
+	 */
+	size_t getHeaderLen() const;
+
+	/**
+	 * Sets missing fields and calculates length

```

Comment:
Maybe we can make this comment a bit more clear? Currently it only sets the protocol version to `0x01`